### ISSUE TITLE
Components not loading asynchronously.

### ISSUE THREAD

**Issue by ryanpcmcquen** [DeloitteDigitalAPAC/react-habitat#28]
I probably have something configured incorrectly, but despite using `registerAsync`:
```js
containerBuilder
    .registerAsync(System.import("./components/Button/Button"))
    .as("Button");
```

My bundle files are the same size regardless of whether I have one component on the page or all of them.

Here's my manifest file: https://github.com/ryanpcmcquen/react-habitat-poc/blob/master/source/Manifest.js

Any ideas what I might have configured incorrectly? Or is this just how `registerAsync` works in this scenario?

**Comment by jennasalau** [comment]
Hi Ryan,

Are you using webpack? My guess is that your `Button` component has a small lines of code number so webpack is being *smart* in this case and saying its better to keep it in the main bundle rather than take on the extra HTTP download overhead.

You can configure this setting using the [MinChunkSizePlugin](https://webpack.js.org/plugins/min-chunk-size-plugin/). For example, I set this to a ridiculous `1` character limit [in our example](https://github.com/DeloitteDigitalAPAC/react-habitat/blob/master/examples/2-code-splitting-with-webpack/webpack.config.js#L26-L28) to force it to create new chunks.

**Comment by ryanpcmcquen** [comment]
@jennasalau, thank you. I have added the `minChunkSizePlugin` but it seems the bundle size still doesn't change between having only the `Card` component (https://ryanpcmcquen.org/react-habitat-poc/card.html):
![screenshot 1516656922](https://user-images.githubusercontent.com/772937/35245798-4ad22cb8-ff79-11e7-9a77-47a3715fb990.png)

And having the `Card`, `Cart`, and `ProductCard` component (https://ryanpcmcquen.org/react-habitat-poc/):
![screenshot 1516656909](https://user-images.githubusercontent.com/772937/35245807-5338a85a-ff79-11e7-91c0-d2cbe680fc20.png)

**Comment by jennasalau** [comment]
Hi Ryan,

Finally had a chance to look into this one. You have indeed found a bug as I can replicate it.

It seems the way we have architected the registrations, the Promise body is evaluating _immediately_ causing the chunks to load upfront. 

Currently we are supporting code separation which was tested, just not lazy loading :(

The problem seems to be with the Habitat API at the very surface level.

```
containerBuilder.registerAsync( new Promise() )
```

The promise executor is evaluating immediately (or in our case resolving the module/chunk).

We will probably have to wrap it in a function that returns a Promise so it only evaluates the executor when called. 

ie

```
containerBuilder.registerAsync(() => { return new Promise() } )
```

Unfortunately this is going to be a breaking change dangit.

**Comment by ryanpcmcquen** [comment]
I'm sad that it will be breaking, but _very_ excited to see this working! It is essential for us. 😸

**Comment by jennasalau** [comment]
I have pre-released this fix as part of v1.0.0-beta (Since we have been sitting stable for a while lets put this as a v1 milestone).

To test this fix please `npm install react-habitat@1.0.0-beta --save-dev` I will be getting our internal project teams to do the same.

**Comment by ryanpcmcquen** [comment]
`npm install react-habitat@1.0.0-beta --save`?

**Comment by jennasalau** [comment]
I guess it depends if you're building a JS library or an app :D

Webpack will put React Habitat into the built binary so no need for the dependency after that.

**Comment by ryanpcmcquen** [comment]
Oh, the README has just `--save`, that is why I asked.

**Comment by jenssogaard** [comment]
@jennasalau just implemented 1.0.0-beta in an existing project and adjusted my syntax. Works as expected 👍

**Comment by ryanpcmcquen** [comment]
Working here too: https://github.com/ryanpcmcquen/react-habitat-poc

**Comment by ryanpcmcquen** [comment]
Is `1.0.0` ready to go stable?

**Comment by jennasalau** [comment]
Yep it's appearing that way. Just need to test typescript and we can go prod.

**Comment by jennasalau** [comment]
Tests complete, this is now published as 1.0.0
